crates/turborepo-filewatch/src/lib.rs

```diff
@@ -271,8 +271,17 @@ async fn watch_events(
                             if event.kind == EventKind::Create(CreateKind::Folder) {
                                 for new_path in &event.paths {
                                     if let Err(err) = manually_add_recursive_watches(new_path, &mut watcher, Some(&broadcast_sender)) {
-                                        warn!("encountered error watching filesystem {}", err);
-                                        break 'outer;
+                                        match err {
+                                            WatchError::WalkDir(_) => {
+                                                // Likely the path no longer exists
```

Comment:
Can we add a debug log here for this scenario?

Reply:
Thanks for taking a look! Added a dbg log statement here https://github.com/vercel/turborepo/pull/9406/commits/f9e08fc34f1925e1176f8e6a64003e713e08db11